server/expression.js

```diff
@@ -204,72 +233,75 @@ expression.getTFAmatLine_ = function(file, name) {
 };
 
 /**
- * Gets the expression matrix profile of a given gene.
+ * Gets the expression matrix profile of given genes and conditions.
  * @param {string} fileExp File name of the expression matrix.
  * @param {string} fileTFA File name of the TFA matrix.
- * @param {string} name Name of the gene to be profiled.
+ * @param {!Array<string>} geneNames Names of the expression matrix.
+ * @param {!Array<string>} conditionNames Names of the expression matrix.
  * @return {?expression.Profile} Gene expression profile as a JS object.
  * @private
  */
-expression.getExpmatLine_ = function(fileExp, fileTFA, name) {
+expression.getExpmatLine_ = function(fileExp, fileTFA, geneNames,
+                                     conditionNames) {
   var bufExp = utils.readFileToBuf(fileExp);
   var bufTFA = null;
   if (fileTFA != null) {
     bufTFA = utils.readFileToBuf(fileTFA);
   }
-
   if (bufExp == null) {
-    console.error('cannot read file', fileExp);
-    return null;
+    return console.error('cannot read file', fileExp), [];
```

Comment:
You merged incorrectly. Your code is having regression. In other words, you didn't take in the refactor's new changes.